contrib/pyln-testing/pyln/testing/utils.py

```diff
@@ -1306,7 +1306,10 @@ def wait_for_onchaind_txs(self, *args):
             # Now grab the corresponding broadcast lightningd did, to get actual tx:
             r = self.daemon.wait_for_log('Broadcast for onchaind tx')
             rawtx = re.search(r'.* tx ([0-9a-fA-F]*)', r).group(1)
-            txid = self.bitcoin.rpc.decoderawtransaction(rawtx, True)['txid']
+            try:
+                txid = self.bitcoin.rpc.decoderawtransaction(rawtx, True)['txid']
+            except Exception as err:
+                raise ValueError(f"Unable to decode rawtx ({rawtx}): {err}")
```

Comment:
I do not think you can get more useful information from this other than the information that you are already getting from the body commit that you added.

I suspect that there is some error with the `txid`, `02000000` is a wrong `txid` right?

```
postdata = '{"version": "1.1", "method": "decoderawtransaction", "params": ["02000000", true], "id": 1}'
```

I think the bug is from the black magic inside the regex `rawtx = re.search(r'.* tx ([0-9a-fA-F]*)', r).group(1)` that I never learn how they works :)